app/views.py

```diff
@@ -7,49 +7,60 @@
 
 _steam_id_re = re.compile('steamcommunity.com/openid/id/(.*?)$')
 
-'''
 @app.before_first_request
 def startup():
-    data = rec.load_file('./training_data')
-    rec.train(data)
-    print('done')
-'''
+    # load from the file.
+    global data, game_matrix
+    # try to load
+    try:
+        print('loading training data')
+        data, game_matrix = rec.load('./training_data')
+    # if something goes wrong, trace a random user and then try again!
+    except Exception:
+        print('no data found, loading random user data')
+        u_info = user_info.get_user_data(76561198045011271)
+        friend_set = user_info.traverse_friend_graph(76561198045011271)
+        for i in friend_set:
+             user_info.get_user_data(i)
+        data, game_matrix = rec.load('./training_data')
+        print('loaded')
+
 
 @app.before_request
 def before_request():
    g.shrek = get_random_shrek()
 
 @app.route('/')
 def index():
-    print(session)
-    print('user' in g)
     if 'user' in session and session['user'] is not None:
         unplayed_games = user_info.get_unplayed_games(session['user'])
-        recs = rec.get_rec(int(session['user']), 10000)
-        #print(recs)
-        naive = False
-        if recs is None and 'naive' not in session:
+        data, game_matrix = rec.load('./training_data')
+        naive =  False
+        # we have already trawled this user, just give them their recs!
```

Comment:
trawling?

Reply:
as in looked at them and their friends
trawl = search thoroughly